deploy/tasks/00-cosmosdb.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Run this task to deploy CosmoDB database and collections:
+ * ts-node cosmosdb.ts
+ * 
+ * This task assumes that the following resources are already created:
+ *  - Resource group
+ *  - CosmoDB database account
+ */
+// tslint:disable:no-console
+// tslint:disable:no-any
+
+import * as config from "./../tfvars.json";
+import { login } from "./login";
+
+import CosmosDBManagementClient = require("azure-arm-cosmosdb");
+import * as documentdb from "documentdb";
+
+const DocumentClient = documentdb.DocumentClient;
+
+const collectionNotExists = (
+  dbClient: documentdb.DocumentClient,
+  dbName: string,
+  collectionName: string
+) => {
+  return new Promise((resolve, reject) => {
+    dbClient.readCollection(
+      documentdb.UriFactory.createDocumentCollectionUri(dbName, collectionName),
+      (err, coll) => {
+        if (err && err.code === 404) {
+          return resolve(true);
+        }
+        reject(coll);
+      }
+    );
+  });
+};
+
+const databaseNotExists = (
+  dbClient: documentdb.DocumentClient,
+  dbName: string
+) => {
+  return new Promise((resolve, reject) => {
+    dbClient.readDatabase(
+      documentdb.UriFactory.createDatabaseUri(dbName),
+      (err, db) => {
+        if (err && err.code === 404) {
+          return resolve(true);
+        }
+        reject(db);
+      }
+    );
+  });
+};
+
+const createDatabaseIfNotExists = (
+  dbClient: documentdb.DocumentClient,
+  dbName: string
+) => {
+  return new Promise((resolve, reject) => {
+    databaseNotExists(dbClient, dbName).then(
+      () => {
+        dbClient.createDatabase({ id: dbName }, (err, ret) => {
+          if (err) {
+            return reject(err);
+          }
+          resolve(ret);
+        });
+      },
+      err => resolve(err)
+    );
+  });
+};
+
+const createCollectionIfNotExists = (
+  dbClient: documentdb.DocumentClient,
+  dbName: string,
+  collectionName: string
+) => {
+  return new Promise((resolve, reject) => {
+    collectionNotExists(dbClient, dbName, collectionName).then(
+      () => {
+        const dbUri = documentdb.UriFactory.createDatabaseUri(dbName);
+        dbClient.createCollection(dbUri, { id: collectionName }, (err, ret) => {
+          if (err) {
+            return reject(err);
+          }
+          resolve(ret);
+        });
+      },
+      err => resolve(err)
+    );
+  });
+};
+
+export const run = async () => {
+  const loginCreds = await login();
+
+  const client = new CosmosDBManagementClient(
+    (loginCreds as any).creds,
+    loginCreds.subscriptionId
+  );
+
+  const databaseAccount = await client.databaseAccounts.get(
+    (config as any).azurerm_resource_group_00,
```

Comment:
perhaps you can define an interface for `config`, otherwise you should check that you have everything you need in `config` before you start provisioning resources (you want to minimize the chance of having the process break while it's running) -> for instance you could split the check in a `plan` task (like Terraform) by printing what resources already exist and what resources the script is going to create

Reply:
I'll add this to TODOs